cylc/flow/scripts/workflow_state.py

```diff
@@ -18,252 +18,402 @@
-        help="Specify a particular status or triggering condition to "
-             f"check for. {conds}{states}",
-        action="store", dest="status", default=None)
+        metavar="STATUS",
+        help=f"Task status. {OPT_DEPR_MSG1}.",
+        action="store", dest="depr_status", default=None)
 
+    # Prior to 8.3.0 --output was just an alias for --message
     parser.add_option(
         "-O", "--output", "-m", "--message",
-        help="Check custom task output by message string or trigger string.",
-        action="store", dest="msg", default=None)
-
-    WorkflowPoller.add_to_cmd_options(parser)
+        metavar="MSG",
+        help=f"Task output message. {OPT_DEPR_MSG2}.",
+        action="store", dest="depr_msg", default=None)
+
+    WorkflowPoller.add_to_cmd_options(
+        parser,
+        d_interval=INTERVAL,
+        d_max_polls=MAX_POLLS
+    )
 
     return parser
 
 
 @cli_function(get_option_parser, remove_opts=["--db"])
-def main(parser: COP, options: 'Values', workflow_id: str) -> None:
-
-    if options.use_task_point and options.cycle:
-        raise InputError(
-            "cannot specify a cycle point and use environment variable")
-
-    if options.use_task_point:
-        if "CYLC_TASK_CYCLE_POINT" not in os.environ:
-            raise InputError("CYLC_TASK_CYCLE_POINT is not defined")
-        options.cycle = os.environ["CYLC_TASK_CYCLE_POINT"]
-
-    if options.offset and not options.cycle:
-        raise InputError(
-            "You must target a cycle point to use an offset")
-
-    # Attempt to apply specified offset to the targeted cycle
-    if options.offset:
-        options.cycle = str(add_offset(options.cycle, options.offset))
-
-    # Exit if both task state and message are to being polled
-    if options.status and options.msg:
-        raise InputError("cannot poll both status and custom output")
-
-    if options.msg and not options.task and not options.cycle:
-        raise InputError("need a taskname and cyclepoint")
-
-    # Exit if an invalid status is requested
-    if (options.status and
-            options.status not in TASK_STATUSES_ORDERED and
-            options.status not in CylcWorkflowDBChecker.STATE_ALIASES):
-        raise InputError(f"invalid status '{options.status}'")
-
-    workflow_id = infer_latest_run_from_id(workflow_id, options.alt_run_dir)
-
-    pollargs = {
-        'workflow_id': workflow_id,
-        'run_dir': get_cylc_run_dir(alt_run_dir=options.alt_run_dir),
-        'task': options.task,
-        'cycle': options.cycle,
-        'status': options.status,
-        'message': options.msg,
-    }
-
-    spoller = WorkflowPoller(
-        "requested state",
-        options.interval,
-        options.max_polls,
-        args=pollargs,
+def main(parser: COP, options: 'Values', *ids: str) -> None:
+
+    if len(ids) != 1:
+        raise InputError("Please give a single ID")
+
+    id_ = ids[0].rstrip('/')  # might get 'id/' due to autcomplete
```

Comment:
Can we use `cylc.flow.id_cli.parse_id` instead?

Reply:
It would be a bit cleaner to use `parse_ids` but the problem is that aborts if the workflow is not found - i.e., if it is not installed yet.

Here we want to be able to start polling before the upstream workflow is installed - it just makes it a bit easier to get a set of interdependent workflows up and running.

I could probably do it by moving ID parsing into the WorkflowPoller class, but it's not an entirely trivial refactor. So given this works as-is, let's leave it for now. I've added a comment in the code about this.